File: 643-maximum-average-subarray-i/maximum-average-subarray-i.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        if( nums.length==0 || nums==null){
            return 0;
        }
        if (nums.length<k){
            return 0;
        }
        int maxi_s=0;
        int n=nums.length;
        for (int i=0; i<k;i++){
            maxi_s+=nums[i];
        }
        int curr_s=maxi_s;
        for(int j=k; j<n;j++){
            curr_s+=nums[j]-nums[j-k];
            maxi_s=Math.max(maxi_s,curr_s);
        }
        return maxi_s/(double)k;
    }
}

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        
        if not nums:
            return 0
        if len(nums)<k:
            return 0
        maxi_s=sum(nums[:k])
        curr_s=maxi_s
        n=len(nums)
        for i in range(k,n):
            curr_s+=nums[i]-nums[i-k]
            maxi_s=max(maxi_s, curr_s)
        return maxi_s/k








        # if not nums:
        #     return 0
        # if len(nums) < k:
        #     return 0
        
        # n = len(nums)
        # max_sum = sum(nums[:k])  
        # current_sum = max_sum
        
        # for i in range(k, n):
        #     current_sum += nums[i] - nums[i - k]  
        #     max_sum = max(max_sum, current_sum)  
        # return max_sum / k 
